src/main/java/seedu/dengue/model/predicate/DeleteDatePredicate.java

```diff
@@ -0,0 +1,65 @@
+package seedu.dengue.model.predicate;
+
+import java.util.Optional;
+
+import seedu.dengue.model.person.ContinuousData;
+import seedu.dengue.model.person.Date;
+import seedu.dengue.model.person.Person;
+import seedu.dengue.model.range.EndDate;
+import seedu.dengue.model.range.Range;
+import seedu.dengue.model.range.StartDate;
+
+/**
+ * Tests that a {@code Person}'s {@code Date} matches the date or date range given.
+ */
+public class DeleteDatePredicate extends PredicateUtil<Person> {
+    public final Optional<Date> date;
+    public final Range<Date> range;
+
+    /**
+     * A class that represents the predicate used to filter the case list.
+     * @param date (Optional) Date input passed in by user.
+     * @param range Range containing (Optional) Date inputs passed in by user.
+     */
+
+    public DeleteDatePredicate(Optional<Date> date, Range<Date> range) {
+        this.date = date;
+        this.range = range;
+    }
+
+    /**
+     * A class that represents the predicate used to filter the case list.
+     * @param date (Optional) Date input passed in by user.
+     */
+
+    public DeleteDatePredicate(Optional<Date> date) {
+        this.date = date;
+        this.range = ContinuousData.generateRange(
+                new StartDate(Optional.empty()),
+                new EndDate(Optional.empty()));
+    }
+
+    /**
+     * A class that represents the predicate used to filter the case list.
+     * @param range Range containing (Optional) Date inputs passed in by user.
+     */
+    public DeleteDatePredicate(Range<Date> range) {
+        this.date = Optional.empty();
+        this.range = range;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        PersonContainsDatePredicate hasDate = new PersonContainsDatePredicate(date);
+        RangeContainsPersonPredicate hasRange = new RangeContainsPersonPredicate(range);
+        return andAll(hasDate, hasRange).test(person);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof DeleteDatePredicate // instanceof handles nulls
+                && date.equals(((DeleteDatePredicate) other).date)
+                && range.equals(((DeleteDatePredicate) other).range));
+    }
+}
```

Comment:
i dont think predicates should be specialised for delete or find functions. the same predicate should be used for find range and delete range

Reply:
My rationale for creating a separate predicate for `delete` is because `delete` can only take in dates and date ranges anyway! Reusing the find predicate would work, but would require one to pass in `Optional.empty()`s for the name, subpostal, etc. arguments each time, which felt a bit messy and unnecessary.

Maybe one way to combine the two while still keeping the code succinct / readable for `delete` would be to have an overloaded constructor in the combined predicate class?

Reply:
I'd personally prefer keeping the two predicates separate for neatness' sake! Even though they have very similar functions, DeleteDatePredicate can specifically only take in date and date range.

I'm not keen on having to create empty Optionals for the name, subpostal, etc. arguments each time I use the predicate for the delete class. I've just tried defining overloaded constructors for the current FindPredicate, but it seems a bit strange to have overloaded constructors just for date and date ranges and not for any of the other parameters of the FindPredicate.

Reply:
Can :D